### PR TITLE
fix: Object spread operator is unnecessarily transformed when used with React

### PR DESCRIPTION
**Pull request by suxin2017**
…th React

**Description:**
fix Object spread operator is unnecessarily transformed when used with React

**BREAKING CHANGE:**

**Related issue (if exists):**
https://github.com/swc-project/swc/issues/7044

### CODE REVIEW

**Review by kdy1 — APPROVED** [review]
Thanks!
You can do `UPDATE=1 cargo test` from `./crates/swc` to update test references.

---


swc-bump:
 - swc_ecma_transforms_react

**Comment by suxin2017** [comment]
> 谢谢！您可以从中更新测试引用。`UPDATE=1 cargo test``./crates/swc`
> 
> SWC凸起：
> 
> * swc_ecma_transforms_react

done

File: crates/swc_ecma_transforms_react/src/jsx/mod.rs [hunk 1]
```diff
@@ -872,13 +870,7 @@ where
...
             Box::new(Expr::Call(CallExpr {
                 span: DUMMY_SP,
-                callee: {
-                    if self.use_builtins {
-                        member_expr!(DUMMY_SP, Object.assign).as_callee()
-                    } else {
-                        helper!(extends, "extends")
-                    }
-                },
+                callee: helper!(extends, "extends"),
```
Comment (kdy1) [inline crates/swc_ecma_transforms_react/src/jsx/mod.rs:873]: I think this should be reverted

File: crates/swc/tests/vercel/loader-only/next-33291/2-classic/output/index.js [hunk 2]
```diff
@@ -50,9 +52,9 @@ export default function Home() {
         var href = _param.href, linkProps = _object_without_properties(_param, [
             "href"
         ]);
-        return React.createElement("link", Object.assign({
+        return React.createElement("link", _object_spread_props(_object_spread({
```
Comment (kdy1) [inline crates/swc/tests/vercel/loader-only/next-33291/2-classic/output/index.js:55]: This looks like a regression.

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thanks!

File: crates/swc_ecma_transforms_react/src/jsx/mod.rs (hunk 1 above)
Comment (suxin2017) [reply crates/swc_ecma_transforms_react/src/jsx/mod.rs:873]: I don't quite understand the difference between configuring use_buildin and use_spread. It looks like the public is the same, only one is {... } One is Object.assign

Comment (suxin2017) [reply crates/swc_ecma_transforms_react/src/jsx/mod.rs:873]: Maybe we can support issue without fixing it, just configure use_spread

**Pull request closed without merging**